catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/GravitinoPaimonTable.java

```diff
@@ -52,6 +57,12 @@ protected TableOperations newOps() {
    */
   public Schema toPaimonTableSchema() {
     Schema.Builder builder = Schema.newBuilder().comment(comment).options(properties);
+    if (partitioning != null) {
+      builder.partitionKeys(
+          Arrays.stream(partitioning)
+              .map(partition -> partition.references()[0].toString())
```

Comment:
could you check the length of `partition.references()`?

Reply:
updated.